src/platform/DefaultPlatformFunctions.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Interface encapsulating various utility functions whose
+ * implementations depend on the platform / JS runtime.
+ */
+class DefaultPlatformFunctions {
+  _UNAME_CACHE: Promise<string> | null;
+
+  constructor() {
+    this._UNAME_CACHE = null as Promise<string> | null;
+  }
+
+  /**
+   * Calls Node's built-in `exec` function.
+   * If the `child_process` module is not available, return an error.
+   */
+  safeExec(
+    cmd: string,
+    cb: (error: unknown, stdout: string | null) => void
+  ): void {
+    cb(new Error('exec not available'), null);
+  }
+
+  getUname(cb: (uname: string) => void): void {
```

Comment:
let's move the entire fancy implementation into `Node**`

Reply:
Moved to `NodePlatformFunctions`. I updated this function to return a promise for better error handling.